### PR TITLE
Sort-Object: more about default properties

### LINKED ISSUE #6382: Default sort properties are insufficiently described
# Documentation Issue

The help page for `Sort-Object` should document that default properties come from `Types.ps1xml` and that if none are provided or defined, the objects themselves will be compared.
## Context of the issue

- URL to the article (Docs or GitHub URL): https://docs.microsoft.com/en-gb/powershell/module/microsoft.powershell.utility/sort-object?view=powershell-7.1&WT.mc_id=ps-gethelp
- Issue affects the following content (check all that apply):

**Conceptual content**
- [ ] Overview and Install
- [ ] Learning PowerShell
  - [ ] PowerShell 101
  - [ ] Deep dives
  - [ ] Remoting
- [ ] Release notes (What's New)
- [ ] Windows PowerShell
  - WMF, ISE, release notes, etc.
- [ ] DSC articles
- [ ] Community resources
- [ ] Sample scripts
- [ ] Gallery articles
- [ ] Scripting and development
  - [ ] Legacy SDK

**Cmdlet reference & about_ topics**
- [X] Version 7.1 preview content
- [X] Version 7.0 content
- [X] Version 6 content
- [X] Version 5.1 content

- Is the issue specific to a platform (Y/N - Win/macOS/Linux): N

## Detailed description of the issue

When I sort objects of type `[DRAWING.SIZE]` by default properties, the result is as if after comparing the strings corresponding to those objects—meaning that `12` comes before `2`.  There is no support for this phenomenon in the current documentation because it does not say what happens when neither explicit nor default properties are available.

### PR DESCRIPTION
**Pull request by yecril71pl**
# PR Summary

I indicated where the default sorting properties come from and what happens if there are still none.  Fixes #6382.

## PR Context

Select the area of the Table of Contents containing the documents being changed.

**Conceptual content**
- [ ] Overview and Install
- [ ] Learning PowerShell
  - [ ] PowerShell 101
  - [ ] Deep dives
  - [ ] Remoting
- [ ] Release notes (What's New)
- [ ] Windows PowerShell
  - WMF, ISE, release notes, etc.
- [ ] DSC articles
- [ ] Community resources
- [ ] Sample scripts
- [ ] Gallery articles
- [ ] Scripting and development
  - [ ] Legacy SDK

**Cmdlet reference & about_ topics**
- [X] Version 7.1 preview content
- [X] Version 7.0 content
- [X] Version 6 content
- [X] Version 5.1 content

## PR Checklist

- [X] I have read the [contributors guide][contrib] and followed the style and process guidelines
- [X] PR has a meaningful title
- [X] PR is targeted at the _staging_ branch
- [X] All relevant versions updated
- [ ] Includes content related to issues and PRs - see [Closing issues using keywords][key].
- [X] This PR is ready to merge and is not **Work in Progress**
  - If the PR is work in progress, please add the prefix `WIP:` or `[WIP]` to the beginning of the
    title and remove the prefix when the PR is ready.

[contrib]: https://docs.microsoft.com/powershell/scripting/community/contributing/overview
[key]: https://help.github.com/en/articles/closing-issues-using-keywords

### CODE REVIEW

File: reference/7.1/Microsoft.PowerShell.Utility/Sort-Object.md [hunk 1]
```diff
@@ -40,7 +40,9 @@ Sort-Object [-Descending] [-Unique] -Bottom <Int32> [-InputObject <PSObject>] [[
 ## DESCRIPTION
 
 The `Sort-Object` cmdlet sorts objects in ascending or descending order based on object property
-values. If sort properties are not included in a command, PowerShell uses default sort properties.
+values. If sort properties are not included in a command, PowerShell uses default sort properties,
+defined as the `DefaultKeyPropertySet` of the type data of the first input object,
+or compares the objects themselves.
```
Comment (sdwheeler) [inline reference/7.1/Microsoft.PowerShell.Utility/Sort-Object.md:45]:
```suggestion
values. If sort properties are not included in a command, PowerShell uses default sort properties defined by the type of the first input object, or compares the objects
themselves.
```

**Comment by sdwheeler** [comment]
@yecril71pl If this change looks OK, please update the other versions.

File: reference/7.1/Microsoft.PowerShell.Utility/Sort-Object.md [hunk 2]
```diff
@@ -82,7 +83,8 @@ The `Get-ChildItem` cmdlet gets the files and subdirectories from the directory
 This command displays the files in the current directory by length in ascending order.
 
 ```
-PS> Get-ChildItem -Path C:\Test -File | Sort-Object -Property Length
+PS> Get-ChildItem -Path C:\Test -File | Sort-Object 
+ty Length
```
Comment (yecril71pl) [inline reference/7.1/Microsoft.PowerShell.Utility/Sort-Object.md:87]: Are you sure?  It looks like a typing accident.

File: reference/7.1/Microsoft.PowerShell.Utility/Sort-Object.md [hunk 3]
```diff
@@ -579,17 +581,23 @@ You can pipe the objects to be sorted to `Sort-Object`.
 ## NOTES
 
 The `Sort-Object` cmdlet sorts objects based on properties specified in the command or the default
-sort properties for the object type. If an object does not have one of the specified properties, the
-property value for that object is interpreted by `Sort-Object` as **Null** and placed at the end of
-the sort order.
+sort properties for the object type. Default sort properties are defined using the
+`DefaultKeyPropertySet` element in a `types.ps1xml` file. For more information, see
```
Comment (yecril71pl) [inline reference/7.1/Microsoft.PowerShell.Utility/Sort-Object.md:585]: using the `PropertySet` named `DefaultKeyPropertySet` in a

File: reference/7.1/Microsoft.PowerShell.Utility/Sort-Object.md [hunk 4]
```diff
@@ -579,17 +581,23 @@ You can pipe the objects to be sorted to `Sort-Object`.
...
 `Sort-Object` uses the **Compare** method for each property. If a property does not implement
 **IComparable**, the cmdlet converts the property value to a string and uses the **Compare** method
-for **System.String**. For more information, see [PSObject.CompareTo(Object) Method](/dotnet/api/system.management.automation.psobject.compareto).
+for **System.String**. For more information, see
+[PSObject.CompareTo(Object) Method](/dotnet/api/system.management.automation.psobject.compareto).
 
 If you sort on an enumerated property such as **Status**, `Sort-Object` sorts by the enumeration
-values. **Stopped** has a value of **1** and **Running** has a value of **4**. **Stopped** is sorted
-before **Running** because of the enumerated values. For more information, see [ServiceControllerStatus](/dotnet/api/system.serviceprocess.servicecontrollerstatus).
+values. For Windows services, **Stopped** has a value of **1** and **Running** has a value of **4**.
```
Comment (yecril71pl) [inline reference/7.1/Microsoft.PowerShell.Utility/Sort-Object.md:598]: Are we supposed to say **Windows** rather than **Microsoft Windows**?

Comment (sdwheeler) [reply reference/7.1/Microsoft.PowerShell.Utility/Sort-Object.md:598]: **Windows** is sufficient here.

File: reference/7.1/Microsoft.PowerShell.Utility/Sort-Object.md (hunk 2 above)
Comment (sdwheeler) [reply reference/7.1/Microsoft.PowerShell.Utility/Sort-Object.md:87]: Thanks. That was an accident.

File: reference/7.1/Microsoft.PowerShell.Utility/Sort-Object.md (hunk 3 above)
Comment (sdwheeler) [reply reference/7.1/Microsoft.PowerShell.Utility/Sort-Object.md]: Fixed it.

File: reference/7.1/Microsoft.PowerShell.Utility/Sort-Object.md (hunk 2 above)
Comment (sdwheeler) [reply reference/7.1/Microsoft.PowerShell.Utility/Sort-Object.md]: Fixed

**Pull request merged**